presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -637,6 +601,44 @@ private void map(Symbol symbol, Symbol canonical)
             mapping.put(symbol, canonical);
         }
 
+        private Map<Symbol, Expression> canonicalize(Map<Symbol, Expression> oldAssignments)
```

Comment:
I don't think it is possible to extract those changes to a separate commit. The reason is that since subquery is unaliased, the assignments have to be unaliased too. Otherwise there will be unreferenced symbols and validator would fail